view/src/components/farms.js

```diff
@@ -358,6 +426,344 @@ class Farms extends Component {
+   * Autocomplete value={}, so separating the two is smoother.
+   */
+  tagsAutocomplete = () => {
+    const { classes } = this.props;
+    const { allFarmTags, tagsQuery } = this.state;
+
+    return (
+      <Autocomplete
+        className={classes.searchBars}
+        multiple
+        id="tags-search"
+        onChange={this.handleTagFilter}
+        options={allFarmTags}
+        value={tagsQuery}
+        fullWidth
+        renderTags={(value, getTagProps) =>
+          value.map((option, index) => (
+            <Chip label={option} {...getTagProps({ index })} />
+          ))
+        }
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            variant="outlined"
+            label="Farm Tags"
+            placeholder="Type or Select a Farm Tag"
+            InputProps={{
+              ...params.InputProps,
+              startAdornment: (
+                <>
+                  <InputAdornment position="start">
+                    <SearchIcon />
+                  </InputAdornment>
+                  {params.InputProps.startAdornment}
+                </>
+              ),
+            }}
+          />
+        )}
+      />
+    );
+  };
+
+  /** Updates filteredData with a new array; used with filters.js queries */
+  updateCards = (newValues) => {
+    if (newValues === []) {
+      return;
+    }
+    this.setState({
+      filteredData: [...newValues],
+    });
+  };
+
+  /** Reset filteredData to the original page data, upon clicking reset */
+  resetCards = () => {
+    this.setState(
+      {
+        filteredData: [...this.state.data],
+        nameQuery: "",
+        locationQuery: "",
+        tagsQuery: [],
+      },
+      this.populateAllFarmTags
+    );
+  };
+
+  /** Renders filtered produce results into Material-UI cards */
+  handleResultsRender = () => {
+    const { classes } = this.props;
+    const { filteredData } = this.state;
+
+    return (
+      <div>
+        <Grid container spacing={2} alignItem="center">
+          {filteredData.map((farm) => (
+            <Grid item xs={12}>
+              <Card className={classes.root} variant="outlined">
+                <CardContent>
+                  <Typography variant="h5" component="h2">
+                    {farm.farmName}
+                  </Typography>
+                  {farm.farmTags.map((tag) => (
+                    <Chip className={classes.chip} label={tag} size="small" />
+                  ))}
+                  <Box
+                    display="flex"
+                    flexDirection="row"
+                    flexWrap="wrap"
+                    p={0}
+                    m={0}
+                  >
+                    <Box p={3}>
+                      <Typography className={classes.pos} color="textSecondary">
+                        Details:
+                      </Typography>
+                      <Typography
+                        variant="body2"
+                        component="p"
```

Comment:
hm, i thought we could do margin/padding instead of m/p?

also, what's this `component="p"` p? can we name that better?

Reply:
@andrewhojel component="p" is in both contact tables for farms and food banks; i'll adjust it in both after confirming with you

Reply:
@andrewhojel ill switch p/m to spelled out versions in foodbanks.js too

Reply:
it seems that "p" is necessary, since it is for HTML <p> -- https://material-ui.com/api/typography/. will keep it as such for now